qiskit/circuit/library/arithmetic/inverse_chebyshev.py

```diff
@@ -0,0 +1,194 @@
+                r_breakpoint = self._a * (5 ** (i + 1))
+            # Calculate the polynomial approximating the function on the current interval
+            poly = Chebyshev.interpolate(lambda x: np.arcsin(self._constant / x), self._degree,
+                                         domain=[self._breakpoints[i], r_breakpoint])
+            # Convert polynomial to the standard basis and rescale it for the rotation gates
+            poly = 2 * poly.convert(kind=np.polynomial.Polynomial).coef
+            # Convert to list and append
+            self._polynomials.append(poly.tolist())
+
+        self._poly_r = PiecewisePolynomialPauliRotations(num_state_qubits, self._breakpoints,
+                                                         self._polynomials)
+
+        self.num_state_qubits = num_state_qubits
+
+    @property
+    def num_ancilla_qubits(self):
+        """Return the number of required ancillas."""
+        return self._poly_r.num_ancilla_qubits
+
+    def _check_configuration(self, raise_on_failure: bool = True) -> bool:
+        valid = True
+
+        if self.num_state_qubits is None:
+            valid = False
+            if raise_on_failure:
+                raise AttributeError('The number of qubits has not been set.')
+
+        if self.num_qubits < self.num_state_qubits + 1:
+            valid = False
+            if raise_on_failure:
+                raise CircuitError('Not enough qubits in the circuit, need at least '
+                                   '{}.'.format(self.num_state_qubits + 1))
+
+        if len(self._breakpoints) != len(self._polynomials):
+            valid = False
+            if raise_on_failure:
+                raise ValueError('Mismatching number of breakpoints and polynomials.')
+
+        return valid
+
+    @property
+    def num_state_qubits(self) -> int:
+        r"""The number of state qubits representing the state :math:`|x\rangle`.
+
+        Returns:
+            The number of state qubits.
+        """
+        return self._num_state_qubits
+
+    @num_state_qubits.setter
+    def num_state_qubits(self, num_state_qubits: Optional[int]) -> None:
+        """Set the number of state qubits.
+
+        Note that this may change the underlying quantum register, if the number of state qubits
+        changes.
+
+        Args:
+            num_state_qubits: The new number of qubits.
+        """
+        if self._num_state_qubits is None or num_state_qubits != self._num_state_qubits:
+            self._invalidate()
+            self._num_state_qubits = num_state_qubits
+
+            self._reset_registers(num_state_qubits)
+
+    def _reset_registers(self, num_state_qubits: Optional[int]) -> None:
+        if num_state_qubits:
+            qr_state = QuantumRegister(num_state_qubits)
+            qr_target = QuantumRegister(1)
+            self.qregs = [qr_state, qr_target]
+
+            if self.num_ancilla_qubits > 0:
+                qr_ancilla = QuantumRegister(self.num_ancilla_qubits)
```

Comment:
(also make sure to import the `AncillaRegister`)
```suggestion
                qr_ancilla = AncillaRegister(self.num_ancilla_qubits)
```